gusto/advection.py

```diff
@@ -300,3 +324,82 @@ def apply(self, x_in, x_out):
         self.q1.assign(x_in)
         self.solver.solve()
         x_out.assign(self.dq)
+
+
+def recovered_apply(self, x_in):
+    """
+    Extra steps to the apply method for the recovered advection scheme.
```

Comment:
I think you could explain what this recovered scheme is for.  Is it a means for obtain a 2nd order accurate advection scheme in the lowest order case?

Reply:
Sure, thanks!